File: internal/contracts/projects.go
package contracts

type ProjectSummary struct {
	ID          string  `json:"id"`
	Name        string  `json:"name"`
	RootDir     string  `json:"rootDir"`
	UpdatedAt   *string `json:"updatedAt"`
	RoutesCount uint32  `json:"routesCount"`
}

type ProjectsOverview struct {
	Active   *ProjectSummary  `json:"active"`
	Archived []ProjectSummary `json:"archived"`
}

type CreateProjectInput struct {
	Name            string  `json:"name"`
	BindHost        string  `json:"bindHost"`
	Port            uint16  `json:"port"`
	UpstreamBaseURL *string `json:"upstreamBaseUrl"`
	ReplaceActive   bool    `json:"replaceActive"`
}

type RestoreProjectRequest struct {
	ID string `json:"id"`
}

type OpenPathRequest struct {
	Path string `json:"path"`
}

type RuntimeTarget struct {
	ProjectID string
	Gateway   GatewaySettings
}

File: frontend/src/lib/contracts/projects.ts
export type ProjectSummary = {
  id: string;
  name: string;
  rootDir: string;
  updatedAt?: string | null;
  routesCount: number;
};

export type ProjectsOverview = {
  active: ProjectSummary | null;
  archived: ProjectSummary[];
};

export type CreateProjectInput = {
  name: string;
  bindHost: string;
  port: number;
  upstreamBaseUrl?: string | null;
  replaceActive?: boolean;
};

export type RestoreProjectRequest = {
  id: string;
};

export type OpenPathRequest = {
  path: string;
};
